### PR TITLE
Add character overlays defined per-emote via char.ini [Overlays]

### PR DESCRIPTION
**Pull request by in1tiate**
overlay images are defined much like emotes are. they do not support (a)/(b) style animations and are meant as static or looping overlay animations to allow for easier creation of characters that utilize such systems

here is an example char.ini using overlays

```ini
[Options]
name=Phoenix
blips=male
side=def

[Emotions]
1 = Normal#pre-normal#normal#0#1

[Overlays]
1 = pre-normal-overlay#normal-overlay
```

### CODE REVIEW

**Comment by in1tiate** [comment]
mind you this feature is heavily experimental, no server software supports this yet

**Comment by in1tiate** [comment]
There's still a nasty bug here causing the overlay widget to not be properly centered during screen shakes, not sure what causes it.

**Comment by in1tiate** [comment]
Alright, as far as I can tell this is now ready for review. It works, even with pairing. Tested this with https://github.com/AttorneyOnline/akashi/pull/40.

**Comment by in1tiate** [comment]
Closing, as I no longer intend to support this

**Comment by Iamgoofball** [comment]
thats a shame, this woulda been useful imo

**Comment by in1tiate** [comment]
Well, mostly it was because another server was using this and I couldn't be assed to support them on a branch I spent like 3 hours on. Given that they self-destructed in the last 30 minutes, I'm willing to give this another go

**Comment by in1tiate** [comment]
That being said, I would need someone to test this for me more extensively as I've only done the bare minimum (does this work on my test server? yes? ok done)

**Comment by Iamgoofball** [comment]
I'll chat w/ the aov content people, I think there's some use cases in this for reducing content size via sharing overlay effects since a lot of characters do have some pretty similar effects sometimes for some emotes.

**Comment by Crystalwarrior** [comment]
Is there a server that implements the networking needed for this feature? If not, can you provide a specification what the server needs to do to pass it over?

File: src/courtroom.cpp [hunk 1]
```diff
@@ -2419,6 +2443,7 @@ void Courtroom::display_pair_character(QString other_charid, QString other_offse
     if (ok && charid > -1) {
       // Show the pair character
       ui_vp_sideplayer_char->show();
+      //ui_vp_sideplayer_char_overlay->show();
```
Comment (oldmud0) [inline src/courtroom.cpp:2446]: ?

**Review by oldmud0 — COMMENTED** [review]
What does this look like in practice? Seems that this enables an additional effect for the preanimation, plus an additional effect for the main animation. What benefit does this have over normal effects, other than being able to specify a default effect for characters plus a special effect for preanims and not polluting the global namespace of overlay effects that we currently have.
My concern would actually be that it wouldn't be flexible enough, e.g. you have an overlay that has slightly different timings than the main emote.

**Comment by stonedDiscord** [comment]
is this gonna make it into 2.10 ?
